android/src/main/java/com/lwansbrough/RCTCamera/RCTCameraModule.java

```diff
@@ -293,44 +510,69 @@ public void hasFlash(ReadableMap options, final Promise promise) {
+            fos.close();
+        } catch (FileNotFoundException e) {
+            return e;
+        } catch (IOException e) {
+            return e;
+        }
+
+        return null;
+    }
+
     private File getOutputMediaFile(int type) {
-        File mediaStorageDir = new File(Environment.getExternalStoragePublicDirectory(
-                Environment.DIRECTORY_PICTURES), "RCTCameraModule");
+        return getOutputFile(
+                type,
+                Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES)
+        );
+    }
+
+    private File getOutputCameraRollFile(int type) {
+        return getOutputFile(
+                type,
+                Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DCIM)
+        );
+    }
 
+    private File getOutputFile(int type, File storageDir) {
         // Create the storage directory if it does not exist
-        if (!mediaStorageDir.exists()) {
-            if (!mediaStorageDir.mkdirs()) {
-                Log.e(TAG, "failed to create directory:" + mediaStorageDir.getAbsolutePath());
+        if (!storageDir.exists()) {
+            if (!storageDir.mkdirs()) {
+                Log.e(TAG, "failed to create directory:" + storageDir.getAbsolutePath());
                 return null;
             }
         }
 
         // Create a media file name
-        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
-        File mediaFile;
+        String photoName = String.format("%s", new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date()));
+
         if (type == MEDIA_TYPE_IMAGE) {
-            mediaFile = new File(mediaStorageDir.getPath() + File.separator +
-                    "IMG_" + timeStamp + ".jpg");
+            photoName = String.format("IMG_%s.jpg", photoName);
         } else if (type == MEDIA_TYPE_VIDEO) {
-            mediaFile = new File(mediaStorageDir.getPath() + File.separator +
-                    "VID_" + timeStamp + ".mp4");
+            photoName = String.format("VID_%s.mp4", photoName);
         } else {
             Log.e(TAG, "Unsupported media type:" + type);
             return null;
         }
-        return mediaFile;
-    }
 
+        return new File(String.format("%s%s%s", storageDir.getPath(), File.separator, photoName));
+    }
 
     private File getTempMediaFile(int type) {
         try {
             String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
-            File outputDir = _reactContext.getCacheDir();
+            File outputDir;
             File outputFile;
 
             if (type == MEDIA_TYPE_IMAGE) {
+                outputDir = _reactContext.getCacheDir(); // for backwards compatibility
                 outputFile = File.createTempFile("IMG_" + timeStamp, ".jpg", outputDir);
             } else if (type == MEDIA_TYPE_VIDEO) {
+                outputDir = _reactContext.getFilesDir(); // for compatibility with react-native-fs
```

Comment:
I agree. @marcejohnson, could you help us with "compatibility with react-native-fs"? These video files will not be cleaned up, as they will be saved into apps private files dir, not temporary dir?